types.go

```diff
@@ -242,20 +242,176 @@ type RaftChanges struct {
 	Leader     raft.ServerID
 }
 
+// VoterEligibility represents whether a node can currently vote,
+// and if it could potentially vote in the future.
+type VoterEligibility struct {
+	currentVoter   bool
+	potentialVoter bool
+}
+
+func (v *VoterEligibility) IsCurrentVoter() bool {
+	return v.currentVoter
+}
+
+func (v *VoterEligibility) IsPotentialVoter() bool {
+	return v.potentialVoter
+}
+
+func (v *VoterEligibility) SetPotentialVoter(isVoter bool) {
+	v.potentialVoter = isVoter
+}
+
+type RaftServerEligibility map[raft.ServerID]*VoterEligibility
+
+// FilterVoters can be used to return all servers that are currently voters,
+// or all servers that are not.
+func (s *RaftServerEligibility) FilterVoters(isCurrentVoter bool) RaftServerEligibility {
```

Comment:
There are a couple things here that are a bit out of the ordinary.  There's nothing wrong with this code, but here's how you could make it less surprising:

```
type RaftServerEligibility struct {
  eligible map[raft.ServerID]*VoterEligibility
}

func (s *RaftServerEligibility) FilterVoters(isCurrentVoter bool) []raft.ServerID
```